sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -156,22 +157,38 @@ object SamplePushDown extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes extra Project added in EliminateSerialization rule.
+ */
+object RemoveExtraProjectForSerialization extends Rule[LogicalPlan] {
```

Comment:
How about we make it more general? e.g. `RemoveAliasOnlyProject`, so that not only object operator can benefit from it.